app/controllers/builds_controller.rb

```diff
@@ -1,6 +1,13 @@
 class BuildsController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :find_project
+  include CurrentProject
+  include ProjectLevelAuthorization
+
+  before_action do
+    find_project(params[:project_id])
+  end
```

Comment:
The params vary depending on the controller. We'll need `params[:project_id]` or `params[:id]`, depending wether we're executing an action against a resource belonging to a project or against the project itself. 

I agree with your solution below, to move this logic into the `CurrenProject#current_project` method, and that's what I'd like to use if not for this particular problem around the different parameters and all the effort from having to change all the ERB files.